src/Promitor.Core/Metrics/MeasuredMetric.cs

```diff
@@ -63,9 +64,14 @@ public static MeasuredMetric CreateForDimension(double? value, string dimensionN
         {
             Guard.NotNullOrWhitespace(dimensionName, nameof(dimensionName));
             Guard.NotNull(timeseries, nameof(timeseries));
-            Guard.For<ArgumentException>(() => timeseries.Metadatavalues.Any() == false);
+            
+            var dimensionMetadataValue = timeseries.Metadatavalues.Where(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);            
+            if(!dimensionMetadataValue.Any())
+            {
+                throw new MissingDimensionException(dimensionName);
+            }
 
-            var dimensionValue = timeseries.Metadatavalues.Single(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);
+            var dimensionValue = dimensionMetadataValue.First();
```

Comment:
Let's use single as we did before
```suggestion
            var dimensionValue = dimensionMetadataValue.Single();
```